### PR TITLE
feat(helm chart): support existing secrets for all dependencies

### PR DESCRIPTION
**Pull request by thomasboni**

### CODE REVIEW

File: charts/r2devops/templates/deployment.yaml [hunk 1]
```diff
@@ -119,19 +121,62 @@ spec:
...
+            - name: JOBS_DB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ $.Values.postgresql.global.postgresql.auth.existingSecret }}"
+                  key: "{{ $.Values.postgresql.global.postgresql.auth.secretKeys.adminPasswordKey }}"
+            {{- else }}
             - name: JOBS_DB_PASSWORD
               value: "{{ $.Values.postgresql.global.postgresql.auth.postgresPassword }}"
+            {{- end }}
+
```
Comment (devpro) [inline charts/r2devops/templates/deployment.yaml:136]: Extra blank line to be removed

File: charts/r2devops/templates/deployment.yaml [hunk 2]
```diff
@@ -119,19 +121,62 @@ spec:
...
+                secretKeyRef:
+                  name: "{{ $.Values.minio.auth.existingSecret }}"
+                  key: "root-password"
+            {{- else }}
             - name: S3_ACCESS_KEY
               value: "{{ $.Values.minio.auth.rootUser }}"
             - name: S3_SECRET_KEY
               value: "{{ $.Values.minio.auth.rootPassword }}"
+            {{- end }}
+
```
Comment (devpro) [inline charts/r2devops/templates/deployment.yaml:156]: Extra blank line to be removed

File: charts/r2devops/templates/deployment.yaml [hunk 3]
```diff
@@ -119,19 +121,62 @@ spec:
               value: "/opt/gitleaks"
             - name: COMPLIANT_LABELS
               value: "Lint,Test,Build,Security,Deploy"
-            - name: JOBS_DB_USER
-              value: "{{ $.Values.postgresql.global.postgresql.auth.username }}"
+
+            # If existing secret is configured for postgresql, use it
```
Comment (devpro) [inline charts/r2devops/templates/deployment.yaml:125]: Comment maybe not really needed :)

File: charts/r2devops/templates/deployment.yaml [hunk 4]
```diff
@@ -119,19 +121,62 @@ spec:
...
+                secretKeyRef:
+                  name: "{{ $.Values.postgresql.global.postgresql.auth.existingSecret }}"
+                  key: "{{ $.Values.postgresql.global.postgresql.auth.secretKeys.adminPasswordKey }}"
+            {{- else }}
             - name: JOBS_DB_PASSWORD
               value: "{{ $.Values.postgresql.global.postgresql.auth.postgresPassword }}"
+            {{- end }}
+
+
+            # If existing secret is configured for object storage, use it
```
Comment (devpro) [inline charts/r2devops/templates/deployment.yaml:138]: Comment maybe not really needed :)

File: charts/r2devops/templates/deployment.yaml [hunk 5]
```diff
@@ -119,19 +121,62 @@ spec:
...
+                  key: "root-password"
+            {{- else }}
             - name: S3_ACCESS_KEY
               value: "{{ $.Values.minio.auth.rootUser }}"
             - name: S3_SECRET_KEY
               value: "{{ $.Values.minio.auth.rootPassword }}"
+            {{- end }}
+
+
+            # If existing secret is configured for redis, use it
```
Comment (devpro) [inline charts/r2devops/templates/deployment.yaml:158]: Comment maybe not really needed :)

**Review by devpro — CHANGES_REQUESTED** [review]
values.yaml and README.md or the documentation must be updated to help the chart user in understanding the logic on the secret values management (either create secrets with this key or pass the values in clear)

File: charts/r2devops/Chart.yaml [hunk 6]
```diff
@@ -2,8 +2,8 @@ apiVersion: v2
 name: r2devops
 description: Helm chart for R2Devops
 type: application
-version: 0.3.8
-appVersion: "1.16.1"
+version: 0.4.0
+appVersion: "1.16.0"
```
Comment (devpro) [inline charts/r2devops/Chart.yaml:6]: Is it normal to lower the appVersion?

Comment (thomasboni) [reply charts/r2devops/Chart.yaml:6]: Nope :) it was a mistake during a rebase

**Comment by thomasboni** [comment]
@devpro Thanks for the review. I've fixed all of your remarks and the documentation update will be merged soon: 

https://gitlab.com/r2devops/documentation/-/merge_requests/90

**Review by devpro — APPROVED** [review]

**Pull request merged**